AndroidSDKGcm/src/main/java/com/leanplum/LeanplumGcmRegistrationJobService.java

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright 2018, Leanplum, Inc. All rights reserved.
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.leanplum;
+
+import android.annotation.TargetApi;
+import android.app.job.JobParameters;
+import android.app.job.JobService;
+import android.content.Intent;
+
+import com.leanplum.internal.Log;
+
+/**
+ * Leanplum GCM registration Job Service for start registration service.
+ *
+ * @author Anna Orlova
+ */
+@TargetApi(21)
+public class LeanplumGcmRegistrationJobService extends JobService {
+  public static final int JOB_ID = -63722755;
+
+  @Override
+  public boolean onStartJob(JobParameters jobParameters) {
+    try {
+      Log.i("GCM InstanceID token needs an update");
```

Comment:
It sounds like the user needs to do something here, maybe more like?
"Updating GCM instanceId token..."